osu.Game/Beatmaps/BeatmapUpdaterMetadataLookup.cs

```diff
@@ -89,21 +93,26 @@ private void lookup(BeatmapSetInfo set, BeatmapInfo beatmapInfo, bool preferOnli
 
                 if (res != null)
                 {
-                    beatmapInfo.Status = res.Status;
+                    beatmapInfo.OnlineID = res.OnlineID;
+                    beatmapInfo.OnlineMD5Hash = res.MD5Hash;
+                    beatmapInfo.LastOnlineUpdate = res.LastUpdated;
 
                     Debug.Assert(beatmapInfo.BeatmapSet != null);
-
-                    beatmapInfo.BeatmapSet.Status = res.BeatmapSet?.Status ?? BeatmapOnlineStatus.None;
                     beatmapInfo.BeatmapSet.OnlineID = res.OnlineBeatmapSetID;
-                    beatmapInfo.BeatmapSet.DateRanked = res.BeatmapSet?.Ranked;
-                    beatmapInfo.BeatmapSet.DateSubmitted = res.BeatmapSet?.Submitted;
-
-                    beatmapInfo.OnlineMD5Hash = res.MD5Hash;
-                    beatmapInfo.LastOnlineUpdate = res.LastUpdated;
 
-                    beatmapInfo.OnlineID = res.OnlineID;
+                    // Some metadata should only be applied if there's no local changes.
+                    if (beatmapInfo.MatchesOnlineVersion || beatmapInfo.Status != BeatmapOnlineStatus.LocallyModified)
+                    {
+                        beatmapInfo.Status = res.Status;
+                        beatmapInfo.Metadata.Author.OnlineID = res.AuthorID;
+                    }
 
-                    beatmapInfo.Metadata.Author.OnlineID = res.AuthorID;
+                    if (beatmapInfo.BeatmapSet.Beatmaps.All(b => b.MatchesOnlineVersion || b.Status != BeatmapOnlineStatus.LocallyModified))
```

Comment:
If I were to suggest anything off the source alone, I'd probably split the `b.MatchesOnlineVersion || b.Status != BeatmapOnlineStatus.LocallyModified` check to a method in this class. I don't think it needs to be something that is more widely available, since it'd be pretty useless outside of this class, but the fact that this same condition shows up four times in one class seems not ideal, I guess?